### PR TITLE
bump parity-scale-codec to 3.6.8

### PR DESCRIPTION
**Pull request by pgherveou**
Bump `parity-scale-codec` to 3.6.8.
Specifically I want to add the fixes to MaxEncodedLen added by https://github.com/paritytech/parity-scale-codec/pull/510

Note: I updated with `cargo upgrade --pinned allow  -p parity-scale-codec@3.6.8 --offline` forcing all versions to 3.6.8
if this undesired (we can update all versions to the compatible version without --pinned allow)

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3590358

**Comment by pgherveou** [comment]
Hum looks like it's breaking the build, will look into it, later today...

**Comment by pgherveou** [comment]
I think that one issue here, is that we have updated parity-scale-codec so that

```rust
#[codec(compact)]
foo: u64,
```

compiles to

```rust
<::codec::Compact<u64> as ::codec::MaxEncodedLen>::max_encoded_len()
```
instead of 
```rust
u64::max_encoded_len()
```

This is all fine but this cause some compilation errors when this field is generic

```rust

#[codec(compact)]
deposit: BalanceOf<T>,

// 53 |         pub deposit: BalanceOf<T, OldCurrency>,
//    |                      ^^^^^^^^^ the trait `parity_scale_codec::MaxEncodedLen` is not implemented for `parity_scale_codec::Compact<<OldCurrency as Currency<<T as frame_system::Config>::AccountId>>::Balance>`

```

**Comment by bkchr** [comment]
@pgherveou I know you have pr to fix this, but did you already yanked this version?

**Comment by pgherveou** [comment]
Yes i did

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3615856

**Comment by ggwpez** [comment]
Hm CI reports `overflow evaluating the requirement parity_scale_codec::Compact<_>: parity_scale_codec::MaxEncodedLen`.

**Comment by pgherveou** [comment]
> evaluating

testing with this patch

also one more question, is that safe to update all crates here, I see that some versions are quite outdated (v3.0.0) is that safe to update ?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3617423

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3617424

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3617413

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-additional-tests
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3617429

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3617425

**Comment by ggwpez** [comment]
> also one more question, is that safe to update all crates here, I see that some versions are quite outdated (v3.0.0) is that safe to update ?

Its semver, so they all use the latest version. But yes to ahead and force them to the latest - makes it easier to read.

**Review by KiChjang — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -528,3 +528,9 @@ wasmi = { opt-level = 3 }
 x25519-dalek = { opt-level = 3 }
 yamux = { opt-level = 3 }
 zeroize = { opt-level = 3 }
+
+
+# patch for crates.io
+[patch.crates-io]
+parity-scale-codec = { git = "https://github.com/paritytech/parity-scale-codec.git", branch = "pg/add_compact_unit_type" }
+
```
Comment (ggwpez) [inline Cargo.toml:536]:
```suggestion
```

Lets try it.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3631487

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3631497

**Comment by pgherveou** [comment]
Need to push 3.6.8 now and try again will do that later

**Comment by ggwpez** [comment]
> Need to push 3.6.8 now and try again will do that later

Okay thanks. The UI tests wont work with a diener patch, since they snapshot the import paths.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3633285

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-additional-tests
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3633290

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3633303

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3633286

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3633284

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3636900

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-tracing
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3636930

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3637038

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3638106

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3638117

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3638374

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3638384

**Comment by pgherveou** [comment]
damned one more, I thought I caught all the tests previously, will yank and try again

**Comment by pgherveou** [comment]
closing for now as the proposed version breaks semver, this patch can wait for 4.x

**Pull request closed without merging**